### PR TITLE
BAU Improve EVCS Client error logging

### PR DESCRIPTION
**Pull request by Joe-Edwards-GDS**
### What changed

- Treat 404 as an expected status code
- Avoid overwriting the log message with the response message
- Include message in `EvcsServiceException`

### Why did it change

EVCS error messages are a bit hard to follow

### CODE REVIEW

File: libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java [hunk 1]
```diff
@@ -197,24 +199,23 @@ private void checkResponseStatusCode(HttpResponse<String> evcsHttpResponse)
...
             }
-            LOGGER.info(
+
+            LOGGER.error(
                     LogHelper.buildLogMessage(
                                     ErrorResponse.RECEIVED_NON_200_RESPONSE_STATUS_CODE
                                             .getMessage())
                             .with(LOG_STATUS_CODE.getFieldName(), statusCode)
-                            .with(LOG_MESSAGE_DESCRIPTION.getFieldName(), responseMessage));
+                            .with(LOG_RESPONSE_MESSAGE.getFieldName(), responseMessage));
```
Comment (Joe-Edwards-GDS) [inline libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java:208]: We shouldn't set `LOG_MESSAGE_DESCRIPTION` - this will overwrite the message above

File: libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java [hunk 2]
```diff
@@ -245,15 +246,19 @@ private HttpResponse<String> sendHttpRequest(HttpRequest evcsHttpRequest)
...
-                    LogHelper.buildErrorMessage(
-                            ErrorResponse.FAILED_AT_EVCS_HTTP_REQUEST_SEND.getMessage(), e));
             if (e instanceof InterruptedException) {
                 // This should never happen running in Lambda as it's single threaded.
                 Thread.currentThread().interrupt();
             } else if (e.getCause() instanceof EvcsServiceException evcsException) {
+                LOGGER.error(
+                        LogHelper.buildErrorMessage(
+                                ErrorResponse.FAILED_AT_EVCS_HTTP_REQUEST_SEND.getMessage(),
+                                evcsException));
```
Comment (Joe-Edwards-GDS) [inline libs/evcs-service/src/main/java/uk/gov/di/ipv/core/library/client/EvcsClient.java:256]: (The underlying exception is likely to have a much more useful description)

File: libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/client/EvcsClientTest.java [hunk 3]
```diff
@@ -191,7 +191,7 @@ void testGetUserVCs_emptyListIsReturned() throws Exception {
     }
 
     @ParameterizedTest
-    @ValueSource(ints = {150, 400})
+    @ValueSource(ints = {400, 500})
```
Comment (Joe-Edwards-GDS) [inline libs/evcs-service/src/test/java/uk/gov/di/ipv/core/library/client/EvcsClientTest.java:194]: IMO we shouldn't worry about 1XX status codes - they're really internal HTTP processing codes that we won't see

**Review by Wynndow — APPROVED** [review]
